enhancements/console/expose-network-features.md

```diff
@@ -0,0 +1,168 @@
+features are accessed via a ConfigMap (read-only permissions) that is written by the Cloud Network
+Operator (CNO). For example, the CNO would write a Config Map with a list of features, e.g.:
+
+```properties
+network_policy_egress=true
+network_policy_peer_ipblock_exceptions=true
+```
+
+or:
+
+```properties
+network_policy_egress=false
+network_policy_peer_ipblock_exceptions=false
+```
+
+The decision of exposing the features as a `string` -> `boolean` map instead of, e.g. an array or
+a set is argued in the [Version Skew Strategy](#version-skew-strategy) section.
+
+### User Stories
+
+* As an Openshift Console developer, I would like to get visibility about the features provided
+  by the CNI type that runs in the customer cluster, so I can conditionally hide network policy fields
+  that do not apply for the user's network type.
+  
+### Implementation Details/Notes/Constraints [optional]
+
+To be decided.
+
+### Risks and Mitigations
+
+To be evaluated.
+
+## Design Details
+
+We will expose the network capabilities as an `sdn-public` Config Map, writeable only by the SDN,
+readable by any `system:authenticated` user.
+
+The [Cluster Network Operator](https://github.com/openshift/cluster-network-operator) will implement
+the logic for exposing the network capabilities.
+
+After the data is accessible, we will modify the `useClusterNetworkFeatures` function
+from the console to load the network features from the previously created config map.
+
+### Open Questions [optional]
+
+N/A
+
+### Test Plan
+
+Unit tests to validate the exposed map values according to the known features of the diverse
+SDN types.
+
+Automated tests for the UI side: check that, according to the selected CNI type, the Network Policy
+creation form will selectively show/hide sections that are supported/unsupported by the CNI.
+
+Integration tests: install the cluster network operator with different CNI types and verify
+that the values of the exposed config maps vary.
+
+QE manual tests to validate the final implementation.
+
+### Graduation Criteria
+
+To be decided.
+
+#### Dev Preview -> Tech Preview
+
+#### Tech Preview -> GA
+
+#### Removing a deprecated feature
+
+### Upgrade / Downgrade Strategy
+
+### Version Skew Strategy
+
+As long as this feature depends on a stable and documented Openshift API feature,
+there are no dependencies that might cause such version skew.
+
+However, as long as we require new features for future versions of the Console,
+it might happen that an outdated "features provider" may not provide information about
+all those _latest features_.
+
+In that case, we may interpret the `string` -> `boolean` features map the following
+way. For each feature `F` in the `features` map:
+
+* If `features[F] === true`, then CNI type supports the feature.
+* If `features[F] === false`, then the CNI type does not support the feature.
+* If `features[F] === undefined`, then the feature' provider does not know
```

Comment:
This _shouldn't_ happen as we're not really asking the CNI to provide this information, the CNO (if it's deploying one of these CNIs _should_ always know what features they support)

Reply:
Unless it's a 3rd party plugin, maybe we don't care as much about those..... i.e https://access.redhat.com/articles/5436171

Reply:
I was thinking on version skew between the CNO and the Console. E.g. a new version of the console that asks for a new network feature but the old version of the CNO does not return anything about its support status.